app/services/visit_request/final_confirmation.rb

```diff
@@ -3,35 +3,20 @@ class FinalConfirmation < ApplicationService
     YES = 'yes'
     NO  = 'no'
 
-    def initialize(visit_request, params)
+    def initialize(visit_request, answer:)
       @visit_request = visit_request
-      @params        = params
+      @answer        = answer
     end
 
     def call
-      return unless schema.call(params).success?
-      return unless visit_request.token == params[:token]
-
-      if yes?
-        visit_request.confirmed!
-      else
-        visit_request.refused!
+      case answer
+      when YES then visit_request.confirmed!
+      when NO  then visit_request.refused!
```

Comment:
Maybe add `else` branch and fail with an error?

Reply:
Makes sense.